### DISCUSSION TITLE
Equivalent of Vim's * to highlight/mark all occurrences of a word?

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [helix-editor/helix#914] [Q&A]
The * key will highlight all occurrences of the word that is under the cursor.

**Comment by Ordoviz** [comment]
Currently, the `*` key sets the search register similarly to Vim and Kakoune. If you then press `%s`, Kakoune provides the contents of the search register as a suggestion, so `*%s<Enter>` is Kakoune's way to select all occurrences of the selected word.

**Reply by archseer** [reply]
Interesting, I should probably auto-populate the prompt with the last search. Right now you can get the same behavior with `*%s`, pressing `up` or `C-p` to load the previous search query (populated by `*`) then `Enter`.

(~~Just spotted a bug, `C-p` populates the prompt but doesn't update the search~~ fixed in 1066b081ddb06a5370371ca139dc8e9b992c1242)

**Comment by Robert-M-Muench** [comment]
The thing is, in Vim, I don't have to mark anything. Just position the cursor, press * → everything marked/highlighted.

It's a super quick one key feature.

**Reply by pickfire** [reply]
But one thing I don't like with vim is that it jumps straight to the next item. I wish it could only just select and highlight first.

**Reply by archseer** [reply]
Because of how multiple selections work with search we actually changed away from this behavior (https://github.com/helix-editor/helix/commit/3426285a6341702ace35817d38cefd2bb8b16437) (https://github.com/helix-editor/helix/issues/713)

Search doesn't actually highlight occurrences in helix, it selects them (and with `N` you can instead append them to your existing selection). This way I can `NNNNNN` and then change all the occurrences at once.

I think with the change in https://github.com/helix-editor/helix/pull/831 you should be able to `%s` `C-s` to replicate this with less keys. In the future https://github.com/helix-editor/helix/pull/589 would allow you to make a single key shortcut.

**Reply by Robert-M-Muench** [reply]
`%s` selects the complete text and I get a `select` prompt. `C-s` assuming this is, `Ctrl+s` doesn't have any effect. Or am I missing something?

And yes, setting up a shortcut if this functionality can be achieved with a multiple-step sequence was my idea too.

**Reply by archseer** [reply]
#831 is not merged yet so C-s doesn't work yet. The shortest example I can think of that works on the current build is miw to select the word under cursor, * to set the word as the search query, %s to select the whole document, arrow up to select the last search query, then enter. That will select all occurrences.

**Reply by Robert-M-Muench** [reply]
This all works, only arrow up doesn't have the desired effect. The selected word is shown, but I have to delete the last char and enter it again to trigger the highlighting.

**Comment by Robert-M-Muench** [comment]
I just tried `vw` (which doesn't select the whole word, if you are not at the beginning, instead from the cursor to the end of the word) and then `N`, which works too. If that sequence could be short-cutted that would mostly do it.

**Comment by morlinbrot** [comment]
I came across this looking for the equivalent of my favourite editor shortcut of all times which is VSCode's `Ctrl+d`.

A simple way to do this in helix is: `miw` (if the word isn't selected yet), `*` to yank into register, followed by `v` (to go into visual mode), then repeating `n` as often as desired.

Bothering me for ages that I didn't have this in nvim so happy to find it in my new favourite editor helix!

**Reply by pooladkhay** [reply]
Thanks that was helpful!
How do get rid of the multiple cursors when the editing is done?

**Reply by lens0021** [reply]
`,` key does
